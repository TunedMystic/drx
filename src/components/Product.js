export default ({ product, addToCart }) => (
    <div className='product' data-testid='product'>
        <h3>{product.name}</h3>
        <p>${product.price}</p>
        <button onClick={() => addToCart(product)}>Add to cart</button>
        <style jsx>{`
            .product {
                display: inline-block;
                padding: 0.5rem;
                width: 190px;
                margin: 10px;

                border: 1px solid black;
            }
        `}</style>
    </div>
)
