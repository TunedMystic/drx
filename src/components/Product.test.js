import { render, fireEvent } from 'react-testing-library';

import Product from './Product';

describe('Component <Product />', () => {
    const addToCart = jest.fn();
    const productData = {
        name: 'Pen',
        price: 3.99
    };
    const productProps = {
        product: productData,
        addToCart
    };

    it('should render correctly', () => {
        const { queryByText, getByTestId } = render(<Product {...productProps} />);
        expect(queryByText('Pen')).toBeInTheDocument();
        expect(queryByText('$3.99')).toBeInTheDocument();
        expect(getByTestId('add-to-cart').textContent).toBe('Add to cart');
    });

    it('should match snapshot', () => {
        const { container } = render(<Product {...productProps} />);
        expect(container).toMatchSnapshot();
    });

    it('should call addToCart on button click', () => {
        const { getByTestId } = render(<Product {...productProps} />);
        fireEvent.click(getByTestId('add-to-cart'));
        // Check that the function was called 1 time.
        expect(addToCart.mock.calls.length).toBe(1);
        // Check that the first function invocation was called with the expected args.
        expect(addToCart.mock.calls[0][0]).toBe(productData);
    });
});
