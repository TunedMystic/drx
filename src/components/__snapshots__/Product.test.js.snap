// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Product /> should match snapshot 1`] = `
<div>
  <div
    class="jsx-891141453 product"
    data-testid="product"
  >
    <h3
      class="jsx-891141453"
    >
      Pen
    </h3>
    <p
      class="jsx-891141453"
    >
      $
      3.99
    </p>
    <button
      class="jsx-891141453 add-to-cart"
      data-testid="add-to-cart"
    >
      Add to cart
    </button>
  </div>
</div>
`;
